Tests/test_file_gif.py

```diff
@@ -507,6 +505,35 @@ def test_save_I(self):
         reloaded = Image.open(out)
         self.assert_image_equal(reloaded.convert('L'), im.convert('L'))
 
+    def test_getdata(self):
+        # test getheader/getdata against legacy values
+        # Create a 'P' image with holes in the palette
+        im = Image._wedge().resize((16,16))
+        im.putpalette(ImagePalette.ImagePalette('RGB'))
+        im.info = {'background':0}
+        
+        passed_palette = bytes(bytearray([255-i//3 for i in range(768)]))
+
+        GifImagePlugin._FORCE_OPTIMIZE = True
+        try:
+            h = GifImagePlugin.getheader(im, passed_palette)
+            d = GifImagePlugin.getdata(im)
+
+            import cPickle as pickle
+            # Enable to get target values on pre-refactor version
+            # with open('Tests/images/gif_header_data.pkl', 'wb') as f:
+            #    f.write(pickle.dumps((h,d)))
+            with open('Tests/images/gif_header_data.pkl', 'rb') as f:
+                (h_target,d_target) = pickle.loads(f.read())
```

Comment:
You need to be careful with the pickle protocol version, pickled files are not necessarily compatible with other Python versions..

Python 2 defaults to version 0:
https://docs.python.org/2/library/pickle.html#data-stream-format

Python 3 defaults to version 3 and is not compatible with Python 2:
https://docs.python.org/3/library/pickle.html#data-stream-format

It'd be safer to explicitly use protocol version 0, 1 or 2 in the save and load.

(Also PEP8 spaces after commas and colons.)